internal/provider/group_resource.go

```diff
@@ -0,0 +1,345 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &GroupResource{}
+	_ resource.ResourceWithConfigure = &GroupResource{}
+)
+
+// NewGroupResource is a helper function to simplify the provider implementation.
+func NewGroupResource() resource.Resource {
+	return &GroupResource{}
+}
+
+type GroupResourceModelInterface interface {
+	ParseHttpResponse(body []byte) error
+	CreateRequestBody() ([]byte, diag.Diagnostics)
+	GetURL() string
+}
+
+// GroupResource is the resource implementation.
+type GroupResource struct {
+	client ProviderHTTPClient
+}
+
+// Metadata returns the resource type name.
+func (r *GroupResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+// Schema defines the schema for the group resource.
+func (r *GroupResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+
+			"inventory_id": schema.Int64Attribute{
+				Required: true,
+			},
+			"name": schema.StringAttribute{
+				Required: true,
+			},
+			"description": schema.StringAttribute{
+				Optional: true,
+			},
+			"group_url": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"variables": schema.StringAttribute{
+				Optional:   true,
+				CustomType: jsontypes.NormalizedType{},
+			},
+		},
+	}
+}
+
+// GroupResourceModel maps the resource schema data.
+type GroupResourceModel struct {
+	InventoryId types.Int64          `tfsdk:"inventory_id"`
+	Name        types.String         `tfsdk:"name"`
+	Description types.String         `tfsdk:"description"`
+	URL         types.String         `tfsdk:"group_url"`
+	Variables   jsontypes.Normalized `tfsdk:"variables"`
+}
+
+func (d *GroupResourceModel) GetURL() string {
+	if IsValueProvided(d.URL) {
+		return d.URL.ValueString()
+	}
+	return ""
+}
+
+func (d *GroupResourceModel) CreateRequestBody() ([]byte, diag.Diagnostics) {
+	body := make(map[string]interface{})
+	var diags diag.Diagnostics
+
+	// Inventory id
+	body["inventory"] = d.InventoryId.ValueInt64()
+
+	// Name
+	body["name"] = d.Name.ValueString()
+
+	// Variables
+	if IsValueProvided(d.Variables) {
+		body["variables"] = d.Variables.ValueString()
+	}
+
+	// URL
+	if IsValueProvided(d.URL) {
+		body["url"] = d.URL.ValueString()
+	}
+
+	// Description
+	if IsValueProvided(d.Description) {
+		body["description"] = d.Description.ValueString()
+	}
+
+	json_raw, err := json.Marshal(body)
+	if err != nil {
+		diags.Append(diag.NewErrorDiagnostic("Body JSON Marshal Error", err.Error()))
+		return nil, diags
+	}
+	return json_raw, diags
+}
+
+func (d *GroupResourceModel) ParseHttpResponse(body []byte) error {
```

Comment:
I think this method should add all of the fields from the response body to the GroupResourceModel, including the inventory id. That way if for some reason the inventory ID in AAP doesn't match what Terraform is expecting, Terraform will error with the appropriate message.